app/mqtt.mjs

```diff
@@ -1,15 +1,264 @@
-import {getReadings} from './modbus.mjs'
+import { getReadings, getDeviceInformation, getSettings, setSetting, getFlagSummary, setFlag } from './modbus.mjs'
 
 const TOPIC_PREFIX = 'eda'
 
-export const getReadingsTopicValues = async (modbusClient) => {
+export const publishValues = async (modbusClient, mqttClient) => {
+    // Create a map from topic name to value that should be published
+    let topicMap = {
+        [`${TOPIC_PREFIX}/status`]: 'online',
+    }
+
+    // Publish state for each mode.
+    const modeSummary = await getFlagSummary(modbusClient)
+
+    for (const [mode, state] of Object.entries(modeSummary)) {
+        const topicName = `${TOPIC_PREFIX}/mode/${mode}`
+
+        // Boolean values are changed to "ON" and "OFF" respectively since those are the
+        // defaults for MQTT switches in Home Assistant
+        topicMap[topicName] = state ? 'ON' : 'OFF'
+    }
+
+    // Publish each reading
     const readings = await getReadings(modbusClient)
-    let topicMap = {}
 
     for (const [reading, value] of Object.entries(readings)) {
         const topicName = `${TOPIC_PREFIX}/readings/${reading}`
-        topicMap[topicName] = value
+        topicMap[topicName] = JSON.stringify(value)
+    }
+
+    // Publish each setting
+    const settings = await getSettings(modbusClient)
+
+    for (const [setting, value] of Object.entries(settings)) {
+        const topicName = `${TOPIC_PREFIX}/settings/${setting}`
+        topicMap[topicName] = JSON.stringify(value)
+    }
+
+    const publishPromises = []
+
+    for (const [topic, value] of Object.entries(topicMap)) {
+        publishPromises.push(mqttClient.publish(topic, value))
+    }
+
+    await Promise.all(publishPromises)
+}
+
+export const subscribeToSettingChanges = async (modbusClient, mqttClient) => {
+    // Subscribe to both settings and mode changes
+    const topicNames = [
+        `${TOPIC_PREFIX}/mode/+/set`,
+        `${TOPIC_PREFIX}/settings/+/set`,
+    ]
+
+    for (const topicName of topicNames) {
+        console.log(`Subscribing to topic(s) ${topicName}`)
+
+        await mqttClient.subscribe(topicName)
+    }
+}
+
+export const handleMessage = async (modbusClient, mqttClient, topicName, payload) => {
+    // Payload looks like a string when logged, but any comparison with === will equal false unless we convert the
+    // Buffer to a string
+    const payloadString = payload.toString()
+
+    console.log(`Received ${payloadString} on topic ${topicName}`)
+
+    // Handle settings updates
+    if (topicName.startsWith('eda/settings/') && topicName.endsWith('/set')) {
+        const settingName = topicName.substring('eda/settings/'.length, topicName.lastIndexOf('/'))
```

Comment:
Same as above, reusing same value over and over again, would make it less error prone to have consts in a common place.